src/Bellatrix.Web/services/DevToolsService.cs

```diff
@@ -28,6 +30,99 @@ public DevToolsService(IWebDriver wrappedDriver)
 
     public DevToolsSessionDomains DevToolsSessionDomains { get; set; }
     public DevToolsSession DevToolsSession { get; set; }
+    public List<NetworkRequestSentEventArgs> RequestsHistory { get; set; }
+    public List<NetworkResponseReceivedEventArgs> ResponsesHistory { get; set; }
+
+    public void StartMonitoringBiDiApi()
+    {
+        RequestsHistory = new();
+        ResponsesHistory = new();
+
+        INetwork networkInterceptor = WrappedDriver.Manage().Network;
+        networkInterceptor.NetworkResponseReceived += (o, s) =>
+        {
+            ResponsesHistory.Add(s);
+        };
+        networkInterceptor.NetworkRequestSent += (o, s) =>
+        {
+            RequestsHistory.Add(s);
+        };
+
+        networkInterceptor.StartMonitoring();
+    }
+
+    public void EmptyResponseAndRequestHistory()
+    {
+        RequestsHistory.Clear();
+        ResponsesHistory.Clear();
+    }
+
+    public List<string> GetSpecificRequestUrls(string requestName)
+    {
+        HashSet<NetworkRequestSentEventArgs> requests = new HashSet<NetworkRequestSentEventArgs>(RequestsHistory);
+        return requests.ToList()
+            .FindAll(r => r.RequestUrl.ToString().Contains(requestName))
+            .Select(fr => fr.RequestUrl).ToList();
+    }
+
+    public void OverrideScreenMetrics(long screenHeight, long screenWidth, bool mobile, double deviceScaleFactor)
+    {
+        var settings = new SetDeviceMetricsOverrideCommandSettings();
+        settings.ScreenHeight = screenHeight;
+        settings.ScreenWidth = screenWidth;
+        settings.Mobile = mobile;
+        settings.DeviceScaleFactor = deviceScaleFactor;
+
+        DevToolsSession.SendCommand(settings);
+    }
+
+    public long GetContentLengthForSpecificResponse(string requestName)
+    {
+        var exactResponse = long.Parse(ResponsesHistory.ToList().Find(r => r.ResponseUrl.Contains(requestName)).ResponseHeaders["content-length"].ToString());
```

Comment:
first get the content lenght then parse it, this will be impossible to debug this way

Reply:
Method refactored to parse after getting the content length